packages/realm-react/CHANGELOG.md

```diff
@@ -4,6 +4,24 @@
 * None
 
 ### Enhancements
+* Added `useProgress` hook which provides a convenient way to access Realm's progress information. It works in a similar way as `realm.addProgressNotification`. **Note: this feature is only intended for flexible sync with the new estimate-based progress notifications.** ([#6797](https://github.com/realm/realm-js/issues/6797))
```

Comment:
The new estimate-based progress notification is supposed to work for both partition-based and flexible sync. We have deprecating the old progress notification in order to provide a unified approach.